src/net.h

```diff
@@ -546,6 +543,20 @@ class CNode
 
     // flood relay
     std::vector<CAddress> vAddrToSend;
+
+    /** Bloom filter to track recent addr messages relayed with this peer.
+     *
+     *  We initialize this filter for outbound peers (other than
+     *  block-relay-only connections) or when an inbound peer sends us an
+     *  address related message (ADDR, ADDRV2, GETADDR, SENDADDRV2).
+     *
+     *  We use the presence of this filter to decide whether a peer is eligible
+     *  for trickle relay of addr messages. This avoids relaying to peers that
+     *  are unlikely to forward them, effectively blackholing self
+     *  announcements. Reasons peers might not support addr relay on the link
+     *  is if they connected to us as a block-relay-only peer or as a light
+     *  client.
+     * */
     std::unique_ptr<CRollingBloomFilter> m_addr_known{nullptr};
```

Comment:
```suggestion
    std::unique_ptr<CRollingBloomFilter> m_addr_known GUARDED_BY(cs_addrKnown){nullptr};
```
and add `cs_addrKnown` as a private member:

```
mutable RecursiveMutex cs_addrKnown;
```

to address the point about race conditions mentioned by @jnewbery

Reply:
Holding off on this to try to figure out if there's a better way I can expose this value through the `getpeerinfo` RPC than having to introduce a mutex for this one hyper-specific use case. Ideas welcome!

Reply:
added `addr_relay_enabled` as an atomic to address this issue.